org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLValidationRootSettings.java

```diff
@@ -0,0 +1,126 @@
+/**
+ *  Copyright (c) 2022 Red Hat Inc. and others.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v2.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *  Red Hat Inc. - initial API and implementation
+ */
+package org.eclipse.lemminx.extensions.contentmodel.settings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * XML validation settings root which defines validation rules for all files.
+ * 
+ * The {@link XMLValidationRootSettings}{@link #filters} can be used to define
+ * custom validation rule for a given file pattern:
+ * 
+ * <code>
+[
+  {
+    "pattern": "**.exsd",
+    "enabled": false
+  },
+  {
+    "pattern": "**{.project,.classpath,plugin.xml,feature.xml,category.xml,.target,.product}",
+    "noGrammar": "ignore"
+  }
+]
+ * </code>
+ * 
+ */
+public class XMLValidationRootSettings extends XMLValidationSettings {
+
+	private static final XMLValidationFilter[] DEFAULT_FILTERS;
+
+	private XMLValidationFilter[] filters;
+
+	static {
+		DEFAULT_FILTERS = createDefaultFilters();
+	}
+
+	public XMLValidationRootSettings() {
+		super();
+		setFilters(DEFAULT_FILTERS);
+	}
+
+	public XMLValidationFilter[] getFilters() {
```

Comment:
I think a javadoc here explaining what the filters are for would be helpful.

Reply:
Fixed.